### PR TITLE
Always push existing subscriptions

### PR DESCRIPTION
**Pull request by julienw**
This goes with https://github.com/fxbox/calendar-server/pull/25

### CODE REVIEW

File: app/js/lib/server/index.js [hunk 1]
```diff
@@ -94,17 +94,14 @@ export default class Server extends EventDispatcher {
...
    */
-  subscribeToNotifications(resubscribe = false) {
+  subscribeToNotifications() {
     if (!this.isLoggedIn) {
       return Promise.reject(new Error(
         'Error while subscribing to push notifications: user is not logged in'
       ));
     }
-    return this[p.webPush].subscribeToNotifications(resubscribe);
+    return this[p.webPush].subscribeToNotifications();
```
Comment (julienw) [inline app/js/lib/server/index.js:105]: not needed anymore

File: app/js/lib/server/webpush.js [hunk 2]
```diff
@@ -31,32 +31,24 @@ export default class WebPush extends EventDispatcher {
...
-          .then((existing) => {
-            if (existing && !resubscribe) {
-              // nothing left to do
-              // @todo We should check if we have this subscription on the
-              // server, if not re-register it on calendar's server.
-              // https://github.com/fxbox/calendar/issues/20
-              return;
-            }
+      .then(
+        (reg) => reg.pushManager.getSubscription()
```
Comment (julienw) [inline app/js/lib/server/webpush.js:35]: actually I'm not even sure this is needed, it seems that "subscribe" actually returns an existing subscription if it exists...

Comment (julienw) [inline app/js/lib/server/webpush.js:35]: I think this is the code in https://cs.chromium.org/chromium/src/content/browser/push_messaging/push_messaging_message_filter.cc?q=PushMessagingHostMsg_Subscribe&sq=package:chromium&dr=C&l=283

That said maybe we should not rely on this, it's just as easy to do what I do here.

File: app/js/lib/server/webpush.js [hunk 3]
```diff
@@ -31,32 +31,24 @@ export default class WebPush extends EventDispatcher {
...
+          .then((existing) =>
+            existing || reg.pushManager.subscribe({ userVisibleOnly: true })
+          )
+      ).then((subscription) =>
+        // The server checks for duplicates
+        this[p.api].post('subscriptions', {
+          subscription,
+          title: `Browser ${navigator.userAgent}`,
+        })
```
Comment (julienw) [inline app/js/lib/server/webpush.js:45]: always posting the subscription

File: app/js/lib/server/webpush.js [hunk 4]
```diff
@@ -31,32 +31,24 @@ export default class WebPush extends EventDispatcher {
...
-              // server, if not re-register it on calendar's server.
-              // https://github.com/fxbox/calendar/issues/20
-              return;
-            }
+      .then(
+        (reg) => reg.pushManager.getSubscription()
+          .then((existing) =>
+            existing || reg.pushManager.subscribe({ userVisibleOnly: true })
+          )
+      ).then((subscription) =>
```
Comment (gmarty) [inline app/js/lib/server/webpush.js]: nit: line break and indentation before `.then(`

File: app/js/lib/server/webpush.js [hunk 5]
```diff
@@ -31,32 +31,24 @@ export default class WebPush extends EventDispatcher {
...
+        // The server checks for duplicates
+        this[p.api].post('subscriptions', {
+          subscription,
+          title: `Browser ${navigator.userAgent}`,
+        })
+      ).catch((error) => {
```
Comment (gmarty) [inline app/js/lib/server/webpush.js]: nit: line break and indentation before `.catch(`

File: app/js/lib/server/webpush.js [hunk 6]
```diff
@@ -31,32 +31,24 @@ export default class WebPush extends EventDispatcher {
...
-          }).catch((error) => {
-            if (Notification.permission === 'denied') {
-              throw new Error('Permission request was denied.');
-            }
-
-            console.error('Error while saving subscription ', error);
-            throw new Error(`Subscription error: ${error}`);
-          });
+        console.error('Error while saving subscription ', error);
+        throw new Error(`Subscription error: ${error}`);
```
Comment (gmarty) [inline app/js/lib/server/webpush.js]: We should either throw or log the error in the console. Can you remove the `console.error` and expand a little the error message?

**Comment by gmarty** [comment]
I know few about push notifications so I trust you on this.
I left a couple of nits and a comments. r+ when they are fixed.

**Pull request merged**